Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/submission-results/SubmissionResults.tsx

```diff
@@ -1,83 +1,38 @@
 import { ITestRunDetailsType } from '../../../hooks/submissions/types';
-import styles from './SubmissionResults.module.scss';
+import TestRunDetails from '../test-run-details/TestRunDetails';
 
 const SubmissionResults = () => {
     const { currentSubmission } = useSubmissionsDetails();
 
-    const getTestRunHeading = (run: ITestRunDetailsType, runIndex: number) => `Test #${runIndex} (${run.resultType}) `;
+    const renderTestRunsDetails = useCallback((testRuns: ITestRunDetailsType[]) => testRuns.map((run, index) => (
+        <TestRunDetails
+          testRun={run}
+          testRunIndex={index + 1}
+        />
+    )), []);
 
-    const getIsCorrectAnswerResultType = (run: ITestRunDetailsType) => run.resultType === 'CorrectAnswer';
-
-    const getTestRunHeadingClassName = useCallback(
-        (run: ITestRunDetailsType) => (getIsCorrectAnswerResultType(run)
-            ? styles.correctTestRunHeading
-            : styles.wrongTestRunHeading),
-        [],
-    );
-
-    const renderTimeAndMemoryUsed = (run: ITestRunDetailsType) => (
-        <>
-            <p>
-                Time used:
-                {run.maxUsedTime}
-                s
-            </p>
-            <p>
-                Memory used:
-                {run.maxUsedMemory}
-                {' '}
-                MB
-            </p>
-        </>
+    const filterRuns = useCallback(
+        (trial: boolean) => currentSubmission!.testRuns.filter((tr) => tr.isTrialTest === trial),
+        [ currentSubmission ],
     );
 
-    const renderTrialTests = useCallback((trialTests: ITestRunDetailsType[]) => trialTests.map((run, index) => (
-        <div className={styles.submissionResultContainer}>
-            <Heading
-              type={HeadingType.secondary}
-              className={(() => getTestRunHeadingClassName(run))()}
-            >
-                {`Zero ${getTestRunHeading(run, index)}`}
-            </Heading>
-            {
-                run.isTrialTest && (run.expectedOutputFragment !== null && run.userOutputFragment !== null)
-                    ? <Diff expectedStr={run.expectedOutputFragment} actualStr={run.userOutputFragment} />
-                    : null
-            }
-            {renderTimeAndMemoryUsed(run)}
-        </div>
-    )), [ getTestRunHeadingClassName ]);
-
-    const renderCompeteTests = useCallback((competeTests: ITestRunDetailsType[]) => competeTests.map((run, index) => (
-        <div className={styles.submissionResultContainer}>
-            <Heading
-              type={HeadingType.secondary}
-              className={(() => getTestRunHeadingClassName(run))()}
-            >
-                {getTestRunHeading(run, index)}
-            </Heading>
-            {renderTimeAndMemoryUsed(run)}
-        </div>
-    )), [ getTestRunHeadingClassName ]);
-
     const renderTestRuns = useCallback(
         () => {
             if (currentSubmission == null) {
                 return null;
             }
```

Comment:
In this function the only use of `currentSubmission` is for null checking. This makes the code a little more difficult to read and follow. The above comment will resolve this, or you can move the check where the variable is actually used.